src/packet/signature/config.rs

```diff
@@ -374,7 +374,7 @@ impl SignatureConfig {
     }
 
     /// Calculate the serialized version of this packet, but only the part relevant for hashing.
-    pub fn hash_signature_data(&self, hasher: &mut dyn std::io::Write) -> Result<usize> {
+    pub fn hash_signature_data(&self, hasher: &mut Box<dyn DynDigest>) -> Result<usize> {
```

Comment:
This effectively reverts https://github.com/rpgp/rpgp/pull/393 :(

We're using that in Signstar for creating OpenPGP signatures from already open hashers and due to us using different version of sha2 crate we can't really plug it in here (since it implements a different `DynDigest` from Rust's perspective).

Would it be possible to revert this small change? If so I'd add a test case so that it doesn't break in the future... (but later, after the dust settles from all these PRs).

Reply:
Changing back to `std::io::Write` for the hashing interface is not gonna be possible in the current state, as `DynDigest` simply does not implement it. So I would suggest for the moment to write a small wrapper proxying things for using another type.